File: translations/codenet/gpt_commented/Go2C/starcoder/Code/s134607050.c


#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// searchRange finds the range of indices in the text where 'R' and 'L' are located.
// It returns the start index, the middle index (first non-'R' character), and the end index (first non-'L' character).
int searchRange(char *text, int start, int limit) {
	int middle = -1;

	// Find the first index where the character is not 'R'
	for (int i = start; i < limit; i++) {
		if (text[i]!= 'R') {
			middle = i;
			break;
		}
	}

	int end = limit;

	// Find the first index after'middle' where the character is not 'L'
	for (int i = middle; i < limit; i++) {
		if (text[i]!= 'L') {
			end = i;
			break;
		}
	}

	// Return the start, middle, and end indices
	return start, middle, end;
}

// answer processes the input text and prints the results based on the 'R' and 'L' characters.
void answer() {
	char *text;
	// Read input text from standard input
	scanf("%s", &text);
	int limit = strlen(text);

	int start = 0;
	int middle = 0;
	int end = 0;
	int firstPrinted = 0;

	// Loop until the end of the text is reached
	while (end < limit) {
		// Get the range of indices for the current segment of 'R's and 'L's
		start, middle, end = searchRange(text, start, limit);
		int len1 = middle - start;
		int len2 = end - middle;

		// Print '0's for the 'R' segment
		for (int i = start; i < middle-1; i++) {
			if (firstPrinted) {
				printf(" ");
			}
			printf("0");
			firstPrinted = 1;
		}

		// Calculate counts based on the lengths of the segments
		int count1 = (len1+1)/2 + len2/2;
		int count2 = len1/2 + (len2+1)/2;

		// Print the counts for the current segment
		if (firstPrinted) {
			printf(" ");
		}
		printf("%d %d", count1, count2);
		firstPrinted = 1;

		// Print '0's for the 'L' segment
		for (int i = middle + 1; i < end; i++) {
			if (firstPrinted) {
				printf(" ");
			}
			printf("0");
			firstPrinted = 1;
		}
		// Move to the next segment
		start = end;
		firstPrinted = 1;
	}
	// Print a newline at the end of the output
	printf("\n");
}

// main function serves as the entry point of the program
int main() {
	answer();
}

File: translations/codenet/uncommented/Go2P/starcoder/Code/s134607050.py
def search_range(text, start, limit):
	middle = -1

	for i in range(start, limit):
		if text[i]!= 'R':
			middle = i
			break

	end = limit

	for i in range(middle, limit):
		if text[i]!= 'L':
			end = i
			break

	return start, middle, end

def answer():
	text = raw_input()
	limit = len(text)

	start = 0
	middle = 0
	end = 0
	first_printed = False

	for end < limit:
		start, middle, end = search_range(text, start, limit)
		len1 = middle - start
		len2 = end - middle

		for i in range(start, middle-1):
			if first_printed:
				print " ",
			print "0",
			first_printed = True

		count1 = (len1+1)/2 + len2/2
		count2 = len1/2 + (len2+1)/2

		if first_printed:
			print " ",
		print count1, count2,
		first_printed = True

		for i in range(middle + 1, end):
			if first_printed:
				print " ",
			print "0",
			first_printed = True
		start = end
		first_printed = True
	print

answer()
# 